go/vt/sqlparser/sql.y

```diff
@@ -1288,39 +1778,70 @@ match_option:
     $$ = QueryExpansionStr
  }
 
+charset:
+  ID
+{
+    $$ = string($1)
+}
+| STRING
+{
+    $$ = string($1)
+}
 
 convert_type:
-  charset
+  BINARY length_opt
```

Comment:
trying to figure out what happened here. it looks like convert_type just totally changed?